src/Command/UnregistrerAdherentsCommand.php

```diff
@@ -0,0 +1,175 @@
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->em = $this->getContainer()->get('doctrine.orm.entity_manager');
+        $this->adherentRepository = $this->em->getRepository(Adherent::class);
+        $this->adherentManager = $this->getContainer()->get('app.adherent_manager');
+        $this->emailNotFound = [];
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $emails = $this->parseCSV($input->getArgument('fileUrl'), 'r');
+        } catch (FileNotFoundException $exception) {
+            $output->writeln(
+                sprintf('%s file not found', $input->getArgument('fileUrl'))
+            );
+
+            return 1;
+        }
+
+        $this->em->beginTransaction();
+
+        if ($this->unregistrateAdherents($emails, $output)) {
+            return 1;
+        }
+
+        $this->em->commit();
+
+        $output->writeln('Unregistration finished.');
+
+        if (0 < count($this->emailNotFound)) {
+            $output->writeln('The following email adress are not found in DB :');
+            foreach ($this->emailNotFound as $email) {
+                $output->writeln($email);
+            }
+        }
+        if (0 < count($this->hosts)) {
+            $output->writeln('The adherents with following email adress are hosts of committees. They are not unregistrated :');
+            foreach ($this->hosts as $email) {
+                $output->writeln($email);
+            }
+        }
+        if (0 < count($this->citizenProjectAdministrators)) {
+            $output->writeln('The adherents with following email adress are administrators of citizen projects. They are not unregistrated :');
+            foreach ($this->citizenProjectAdministrators as $email) {
+                $output->writeln($email);
+            }
+        }
+    }
+
+    private function parseCSV(string $filename): array
+    {
+        $emails = [];
+        if (false === ($handle = fopen($filename, 'r'))) {
+            throw new FileNotFoundException(sprintf('% not found', $filename));
+        }
+
+        $isFirstRow = true;
+        while (false !== ($data = fgetcsv($handle, 100000, ';'))) {
+            if (true === $isFirstRow) {
+                $isFirstRow = false;
+
+                continue;
+            }
+
+            $row = array_map('trim', $data);
+            $emails[] = $row[4];
+        }
+        fclose($handle);
+
+        return $emails;
+    }
+
+    private function unregistrateAdherents(array $emails, OutputInterface $output): bool
+    {
+        $batchSize = 20;
+        $i = 1;
+        $progress = new ProgressBar($output, count($emails));
+        $progress->start();
+
+        foreach ($emails as $email) {
+            try {
+                if (null === ($adherent = $this->adherentRepository->findByEmail($email))) {
```

Comment:
Not related, but this method is badly named, should be called `findOneByEmail` to respect Doctrine conventions and avoid confusion.

Reply:
Done